DrawEngine/DrawEngine.cs

```diff
@@ -0,0 +1,409 @@
+            }
+            Console.WriteLine();
+        }        
+        public void PrintTopBook(BookLength lengthBooks)
+        {                               
+            
+            for (int i = 0; i <= lengthBooks.maxStringLength + 3; i++)
+            {
+                Console.Write("-");
+            }
+            Console.WriteLine();
+            Console.Write("|");            
+
+            int dynamicIndent = lengthBooks.maxBookIDLength; // Динамічний відступ, який можна змінювати
+            string text = "Book ID";
+            Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+
+            dynamicIndent = lengthBooks.maxTitleLength; // Динамічний відступ, який можна змінювати
+            text = "Title";
+            Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+
+            dynamicIndent = lengthBooks.maxAuthorLength; // Динамічний відступ, який можна змінювати
+            text = "Author";
+            Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+            Console.WriteLine();
+        }
+
+        public void PrintMiddleBook(BookLength lengthBooks)
+        {            
+            var books = _booksRepository.GetAll();
+            
+            foreach (var book in books)
+            {
+                PrintLowBook(lengthBooks);
+                Console.Write("|");                
+
+                if (book == null)
+                    throw new Exception("book not found");
+
+                int dynamicIndent = lengthBooks.maxBookIDLength; // Динамічний відступ, який можна змінювати
+                string text = Convert.ToString(book.Book_ID);
+                Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+
+                dynamicIndent = lengthBooks.maxTitleLength; // Динамічний відступ, який можна змінювати
+                text = Convert.ToString(book.Title); ;
+                Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+
+                dynamicIndent = lengthBooks.maxAuthorLength; // Динамічний відступ, який можна змінювати
+                text = Convert.ToString(book.Author); ;
+                Console.Write(text + new string(' ', dynamicIndent - text.Length) + "|");
+
+                Console.WriteLine();
+            }
+
+        }
+        public void PrintLowBook(BookLength lengthBooks)
+        {                     
+            for (int i = 0; i <= lengthBooks.maxStringLength + 3; i++)
+            {
+                Console.Write("-");
+            }
+            Console.WriteLine();
+        }
+        public void PrintAllSales()
+        {            
+            PrintTopSale(CalculateSaleLength());
+            PrintMiddleSale(CalculateSaleLength());
+            PrintLowSale(CalculateSaleLength());
+        }
+        public void PrintAllBooks()
+        {            
+            PrintTopBook(CalculateBookLength());
```

Comment:
call once CalculateBookLength() and set to variable - then transmit this variable to method as parameter

Reply:
done